splice_machine/src/main/java/com/splicemachine/derby/utils/ResultHelper.java

```diff
@@ -0,0 +1,194 @@
+        row = null;
+    }
+
+    public ResultSet getResultSet() throws SQLException {
+        finishRow();
+
+        EmbedConnection conn = (EmbedConnection) BaseAdminProcedures.getDefaultConn();
+        Activation lastActivation = conn.getLanguageConnection().getLastActivation();
+
+        IteratorNoPutResultSet resultsToWrap = new IteratorNoPutResultSet(rows, getColumnDescriptorsArray(), lastActivation);
+        try {
+            resultsToWrap.openCore();
+        } catch (StandardException se) {
+            throw PublicAPI.wrapStandardException(se);
+        }
+        return new EmbedResultSet40(conn, resultsToWrap, false, null, true);
+    }
+
+    class Column {
+        public int index;
+        boolean set = false;
+
+        public void finishRow() {
+            if( !set ) init();
+            set = false;
+        }
+
+        void add(String name, int length)
+        {
+            columnDescriptors.add( new GenericColumnDescriptor(name, create(length)) );
+            index = columnDescriptors.size();
+            columns.add(this);
+        }
+
+        void init() { throw new RuntimeException("not implemented"); }
+
+        public DataTypeDescriptor create(int length) { throw new RuntimeException("not implemented"); }
+    }
+
+    class VarcharColumn extends Column {
+        @Override
+        public DataTypeDescriptor create(int length)
+        {
+            return DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, length);
+        }
+        public void set(String value) {
+            assert row != null;
+            row.setColumn(index, new SQLVarchar(value));
+            set = true;
+        }
+        @Override
+        public void init() {
+            set("");
+        }
+    }
+    class BigintColumn extends Column {
+        @Override
+        public DataTypeDescriptor create(int length)
+        {
+            return DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.BIGINT, length);
+        }
+        public void set(long value) {
+            assert row != null;
+            row.setColumn(index, new SQLLongint(value));
+            set = true;
+        }
+        @Override
+        public void init() {
+            set(0);
+        }
+    }
+
+    class TimestampColumn extends Column {
+        @Override
+        public DataTypeDescriptor create(int length)
+        {
+            return DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.TIMESTAMP, length);
+        }
+        public void set(org.joda.time.DateTime value) throws StandardException {
+            assert row != null;
+            row.setColumn(index, new SQLTimestamp(value));
+            set = true;
+        }
+        @Override
+        public void init() {
+            try {
+                set( new DateTime(0));
+            } catch (StandardException e) {
+            }
+        }
+    }
+    private List<ExecRow> rows = new ArrayList<>();
+    private ArrayList<Column> columns = new ArrayList<>();
```

Comment:
better to use interface List in the variable definition, like rows, private List<Column> columns = new ArrayList<>();

Reply:
done